catalog/nsili/catalog-nsili-endpoint/src/main/java/com/connexta/alliance/nsili/endpoint/managers/AccessManagerImpl.java

```diff
@@ -0,0 +1,237 @@
+        return NUM_PRIORITIES;
+    }
+
+    @Override
+    public SetAvailabilityRequest set_availability(Product[] products,
+            AvailabilityRequirement availability_requirement, String use_mode, short priority)
+            throws InvalidInputParameter, ProcessingFault, SystemFault {
+        return new _SetAvailabilityRequestStub();
+    }
+
+    @Override
+    public Request[] get_active_requests() throws ProcessingFault, SystemFault {
+        return new Request[0];
+    }
+
+    @Override
+    public int get_default_timeout() throws ProcessingFault, SystemFault {
+        return defaultTimeout;
+    }
+
+    @Override
+    public void set_default_timeout(int new_default)
+            throws InvalidInputParameter, ProcessingFault, SystemFault {
+        this.defaultTimeout = new_default;
+    }
+
+    @Override
+    public int get_timeout(Request aRequest)
+            throws InvalidInputParameter, ProcessingFault, SystemFault {
+        return defaultTimeout;
+    }
+
+    @Override
+    public void set_timeout(Request aRequest, int new_lifetime)
+            throws InvalidInputParameter, ProcessingFault, SystemFault {
+
+    }
+
+    @Override
+    public void delete_request(Request aRequest)
+            throws InvalidInputParameter, ProcessingFault, SystemFault {
+
+    }
+
+    public String getProductId(Product product)
+            throws UnsupportedEncodingException, WrongPolicy, WrongAdapter {
+        byte[] productOidBytes = _poa().reference_to_id(product);
+        return new String(productOidBytes, NsiliEndpoint.ENCODING);
+    }
+
+    public Metacard getMetacard(String id) {
+        Metacard metacard = null;
+        List<Result> results = new ArrayList<>();
+
+        Filter filter = filterBuilder.attribute(Metacard.ID).is().equalTo().text(id);
+
+        QueryImpl catalogQuery = new QueryImpl(filter);
+        catalogQuery.setRequestsTotalResultsCount(false);
+        catalogQuery.setPageSize(10);
+
+        QueryRequestImpl catalogQueryRequest = new QueryRequestImpl(catalogQuery);
+
+        try {
+            QueryResultsCallable
+                    queryCallable = new QueryResultsCallable(catalogQueryRequest);
+            results.addAll(subject.execute(queryCallable));
+
+        } catch (Exception e) {
+            LOGGER.warn("Unable to query catalog", e);
+        }
+
+        if (!results.isEmpty()) {
+            metacard = results.iterator().next().getMetacard();
+        }
+
+        return metacard;
+    }
+
+    public boolean isUrlValid(String urlStr) {
+        if (urlStr != null) {
+            try {
+                URL url = new URL(urlStr);
+                URLConnection urlConnection = url.openConnection();
+                urlConnection.connect();
+
+                int contentLength = urlConnection.getContentLength();
+                if (contentLength > 0) {
+                    return true;
+                }
+            } catch (IOException ignore) {}
```

Comment:
If we get an IOException when attempting to access the URL it isn't valid for the purpose of this method, so we can ignore the exception as the method defaults to returning false.